src/Task/Archive/Extract.php

```diff
@@ -0,0 +1,196 @@
+<?php
+namespace Robo\Task\Archive;
+
+use Robo\Result;
+use Robo\Task\BaseTask;
+
+/**
+ * Extracts an archive.
+ *
+ * Note that often, distributions are packaged in tar or zip archives
+ * where the topmost folder may contain variable information, such as
+ * the release date, or the version of the package.  This information
+ * is very useful when unpacking by hand, but arbitrarily-named directories
+ * are much less useful to scripts.  Therefore, by default, Extract will
+ * remove the top-level directory, and instead store all extracted files
+ * into the directory specified by $archivePath.
+ *
+ * To keep the top-level directory when extracting, use
+ * `preserveTopDirectory(true)`.
+ *
+ * ``` php
+ * <?php
+ * $this->taskExtract($archivePath)
+ *  ->to($destination)
+ *  ->preserveTopDirectory(false) // the default
+ *  ->run();
+ * ?>
+ * ```
+ *
+ * @method to(string) location to store extracted files
+ */
+class Extract extends BaseTask
+{
+    use \Robo\Common\DynamicParams;
+    use \Robo\Common\Timer;
+    use \Robo\Common\PHPStatus;
+
+    protected $filename;
+    protected $to;
+    private $preserveTopDirectory = false;
+
+    public function __construct($filename)
+    {
+        $this->filename = $filename;
+    }
+
+    function run()
+    {
+        if (!file_exists($this->filename)) {
+            $this->printTaskError("File {$this->filename} does not exist");
+            return false;
+        }
+        if (!($mimetype = static::archiveType($this->filename))) {
+            $this->printTaskError("Could not determine type of archive for {$this->filename}");
+            return false;
+        }
+
+        // We will first extract to $extractLocation and then move to $this->to
+        $extractLocation = static::getTmpDir();
+        @mkdir($extractLocation);
+        @mkdir(dirname($this->to));
+
+        $this->startTimer();
+
+        $this->printTaskInfo("Extracting <info>{$this->filename}</info>");
+        // Perform the extraction of a zip file.
+        if (($mimetype == 'application/zip') || ($mimetype == 'application/x-zip')) {
+            $result = $this->extractZip($extractLocation);
+        }
```

Comment:
Not sure what Nitpick-CI is trying to say here; a space should not come after a closing brace, newline is correct. Also, this line is the same as all of the other similar occurrences of this pattern.